graph/connected_components_with_bfs.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ *
+ * \file
+ * \brief [Graph Connected Components
+ * (Connected Components)]
+ * (https://en.wikipedia.org/wiki/Component_(graph_theory))
+ *
+ * \author [Ritik Rathor](https://github.com/ritik-369)
+ *
+ * \details
+ * A graph is a collection of nodes also called vertices and these vertices
+ * are connected by edges. A connected component in a graph refers to a set of
+ * vertices which are reachable form one another.
+ *
+ * <pre>
+ * Example - Here is graph with 3 connected components
+ *
+ *      1   4           5               8
+ *     / \ /           / \             / \
+ *    2---3           6   7           9   10
+ *
+ *    first          second           third
+ *    component      component        component
+ * </pre>
+ *
+ */
+ 
+
+//                  WE WILL LEARN CALCULATING NO. OF CONNECTED COMPONENTS IN GRAPH USING BFS BY SOLVING A HACKEREARTH QUESTION
+//                  LINK=   https://www.hackerearth.com/problem/algorithm/connected-components-in-a-graph/
+
+
+#include <iostream>  // for IO operations
+#include <vector>    //for using vectors to make graph or adjacency list
+#include <queue>     //for using queue data structure for pushing and poping nodes during bfs
+
```

Comment:
You may need to update the function names inside the namespace.
```suggestion
/**
 * @namespace graph
 * @brief Graphical algorithms
 */
namespace graph {
```